common/scala/src/main/scala/org/apache/openwhisk/core/containerpool/Container.scala

```diff
@@ -73,6 +73,10 @@ trait Container {
   /** HTTP connection to the container, will be lazily established by callContainer */
   protected var httpConnection: Option[ContainerClient] = None
 
+  /** maxConcurrent+timeout are cached during first init, so that resuming connections can reference */
+  protected var containerHttpMaxConcurrent: Int = 1
+  protected var containerHttpTimeout: FiniteDuration = 60.seconds
```

Comment:
I know this was the case before, but we should add a comment to the trait stating that it's not thread-safe and MUST the caller must ensure synchronization (which we do by using an actor)

Reply:
@markusthoemmes Hello. I'm a newbie at scala and having a hard time reading the source code. Can you please explain what thread-safe means above, and why resume() (which I'm guessing that it somehow calls "docker container resume") is not thread safe??